src/awkward1/operations/convert.py

```diff
@@ -1139,6 +1139,180 @@ def getfunction(layout, depth):
     else:
         return out
 
+def to_arrow(layout):
+    import pyarrow
+
+    def recurse(layout, mask = None):
+
+        if isinstance(layout, awkward1.layout.NumpyArray):
+            numpy_arr = numpy.asarray(layout)
+            if(numpy_arr.ndim == 1):
+                return pyarrow.array(numpy_arr, mask=mask)
+            else:
+                return pyarrow.Tensor.from_numpy(numpy_arr)
+        
+        elif isinstance(layout, awkward1.layout.EmptyArray):
+            return pyarrow.array(numpy.array([], dtype=numpy.float64))
+
+        elif isinstance(layout, (awkward1.layout.IndexU8, 
+                                awkward1.layout.IndexU32, 
+                                awkward1.layout.Index8, 
+                                awkward1.layout.Index32, 
+                                awkward1.layout.Index64)):
+            return pyarrow.array(numpy.asarray(layout))
+        
+        elif isinstance(layout, awkward1.layout.ListOffsetArray32):
+
+            if (awkward1.operations.describe.parameters(layout).get("__array__") == "bytestring"):
+                arr = [awkward1.behaviors.string.ByteBehavior(
+                    layout[i]).__bytes__() for i in range(len(layout))]
+                if mask is not None:
+                    return pyarrow.array(arr, mask=numpy.array(mask, dtype=numpy.bool))
+                else:
+                    return pyarrow.array(arr)
+
+            if (awkward1.operations.describe.parameters(layout).get("__array__") == "string"):
+                arr = [awkward1.behaviors.string.CharBehavior(
+                    layout[i]).__str__() for i in range(len(layout))]
+                if mask is not None:
+                    return pyarrow.array(arr, mask=numpy.array(mask, dtype=numpy.bool))
+                else:
+                    return pyarrow.array(arr)
+            
+            if mask is not None:
+                indices = []
+                for i in range(len(mask)):
+                    if mask[i] == 1:
+                        indices.append(None)
+                    else:
+                        indices.append(i)
```

Comment:
This could go to a helper function.

Reply:
Okay.

Reply:
This is done, I need to update the other conditions to accommodate for this.